rust/crd/src/lib.rs

```diff
@@ -172,11 +182,28 @@ impl KafkaCluster {
             .map(|memory_limit| to_java_heap(memory_limit, JVM_HEAP_FACTOR))
             .transpose()
     }
-}
 
-#[derive(Debug, Snafu)]
-#[snafu(display("object has no namespace associated"))]
-pub struct NoNamespaceError;
+    /// Returns the provided docker image e.g. 2.8.1-stackable0.1.0
+    pub fn image_version(&self) -> Result<&str, Error> {
+        self.spec
+            .version
+            .as_deref()
+            .context(ObjectHasNoVersionSnafu)
+    }
+
+    /// Returns our semver representation for product config e.g. 2.8.1
+    pub fn product_version(&self) -> Result<&str, Error> {
+        let image_version = self.image_version()?;
+        image_version
+            .split('-')
+            .collect::<Vec<_>>()
+            .first()
+            .cloned()
```

Comment:
NIT, removes collecting whole list and throwing it away later on
```suggestion
            .next()
```

Reply:
Thanks, done!